modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -1403,7 +1405,25 @@ private Publisher<BinaryRow> scan(
                         columnsToInclude,
                         implicit
                 ),
-                (commit, fut) -> postEnlist(fut, commit, actualTx, implicit && !commit)
+                (intentionallyClose, fut) -> {
+                    CompletableFuture<Void> opFut;
+
+                    if (implicit) {
+                        opFut = fut.thenApply(cursorId -> null);
+                    } else {
+                        var replicationGrpId = new TablePartitionId(tableId, partId);
+
+                        opFut = tx.enlistedNodeAndTerm(replicationGrpId) != null ? completeScan(
+                                tx.id(),
+                                replicationGrpId,
+                                fut,
+                                tx.enlistedNodeAndTerm(replicationGrpId).get1(),
+                                intentionallyClose
+                        ) : fut.thenApply(cursorId -> null);
+                    }
+
+                    return postEnlist(opFut, intentionallyClose, actualTx, implicit && !intentionallyClose);
```

Comment:
Did you check that intentionallyClose is calculated correctly? I mean not this specific place, but everywhere possible.